test/remote-models.test.js

```diff
@@ -11,25 +11,97 @@ var TaskEmitter = require('strong-task-emitter');
 
 describe('Remote model tests', function() {
   var ctx = this;
-
+  var memoryDs = helper.createMemoryDataSource();
   beforeEach(function(done) {
+    memoryDs.connector.cache = {}; // empty cache of memory connector
```

Comment:
This is smelly, we should be creating a new datasource for each test instead. I tried that change myself but then some of your tests started to fail with no clear reason. I think this is a problem of sharing the same model registry between the server and the client (remote) app. I opened a pull request to address that - see https://github.com/strongloop/loopback-connector-remote/pull/82. Let's land it first and then I can help you to rework this patch to use the new approach.

Reply:
Will have to check... After #82 the change should be quite straight forward